assert/assertions.go

```diff
@@ -263,6 +273,24 @@ func CallerInfo() []string {
 			isTest(name, "Example") {
 			break
 		}
+
+		if more {
+			continue
+		}
+		// We know we already have less than a buffer's worth of frames
+		if !maybeMore {
+			break
+		}
+		offset += stackFrameBufferSize
+		n = runtime.Callers(offset, pcs)
+		if n == 0 {
+			break
+		}
+
+		maybeMore = n == stackFrameBufferSize
```

Comment:
Copying this comment from #1615 which also contains this change.

I'm not a fan of `more` and `maybeMore`, this code is actually two nested loops and I think it would be more readable if you coded it that way. I think that pure behaviour is usually better than data (varibles) used only to alter behaviour.

Reply:
I just pushed a commit that moves this to two loops (though still using `more` to `break` out of the inner loop). LMK if that's along the lines of what you were thinking.